src/psyclone/psyir/nodes/omp_directives.py

```diff
@@ -230,6 +241,833 @@ class OMPSerialDirective(OMPRegionDirective, metaclass=abc.ABCMeta):
+        return (arraymixin1.is_full_range(index) and
+                arraymixin2.is_full_range(index))
+
+    def _compute_accesses_get_start_stop_step(self, preceding_nodes, task,
+                                              symbol):
+        '''
+        Computes the start, stop and step values used in the _compute_accesses
+        function by searching through the preceding nodes for the last access
+        to the symbol.
+
+        :param preceding_nodes: a list of nodes that precede the task in the
+                                tree.
+        :type preceding_nodes: List[:py:class:`psyclone.psyir.nodes.Node`]
+        :param task: the OMPTaskDirective node being used in
+                     _compute_accesses.
+        :type task: :py:class:`psyclone.psyir.nodes.OMPTaskDirective`
+        :param symbol: the symbol used by the ref in _compute_accesses.
+        :type symbol: :py:class:`psyclone.psyir.symbols.Symbol`
+
+        :returns: a tuple containing the start, stop and step nodes (or None
+                  if there is no value).
+        :rtype: Tuple[:py:class`psyclone.psyir.nodes.Node, None]
+        '''
+        start = None
+        stop = None
+        step = None
+        for node in preceding_nodes:
+            # Only Assignment, Loop or Call nodes can modify the symbol in our
+            # Reference
+            if not isinstance(node, (Assignment, Loop, Call)):
+                continue
+            # At the moment we allow all IntrinsicCall nodes through, and
+            # assume that all IntrinsicCall nodes we find don't modify
+            # symbols, but only read from them.
+            if isinstance(node, Call) and not isinstance(node, IntrinsicCall):
+                # Currently opting to fail on any non-intrinsic Call.
+                # Potentially it might be possible to check if the Symbol is
+                # written to and only if so then raise an error
+                raise UnresolvedDependencyError(
+                        "Found a Call in preceding_nodes, which "
+                        "is not yet supported.")
+            if isinstance(node, Assignment) and node.lhs.symbol == symbol:
+                start = node.rhs.copy()
+                break
+            if isinstance(node, Loop) and node.variable == symbol:
+                # If the loop is not an ancestor of the task then
+                # we don't currently support it.
+                ancestor_loop = task.ancestor(Loop, limit=self)
+                is_ancestor = False
+                while ancestor_loop is not None:
+                    if ancestor_loop == node:
+                        is_ancestor = True
+                        break
+                    ancestor_loop = ancestor_loop.ancestor(Loop, limit=self)
+                if not is_ancestor:
+                    raise UnresolvedDependencyError(
+                            "Found a dependency index that "
```

Comment:
Please include the variable name in this error message.

Reply:
Added

Reply:
Done